helix-term/src/ui/picker/handlers.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    path::Path,
+    sync::{atomic, Arc},
+    time::Duration,
+};
+
+use helix_event::AsyncHook;
+use tokio::time::Instant;
+
+use crate::ui::overlay::Overlay;
+
+use super::{CachedPreview, DynQueryCallback, Picker};
+
+pub(super) struct PreviewHighlightHandler<T: 'static + Send + Sync, D: 'static + Send + Sync> {
+    trigger: Option<Arc<Path>>,
+    phantom_data: std::marker::PhantomData<(T, D)>,
+}
+
+impl<T: 'static + Send + Sync, D: 'static + Send + Sync> Default for PreviewHighlightHandler<T, D> {
+    fn default() -> Self {
+        Self {
+            trigger: None,
+            phantom_data: Default::default(),
+        }
+    }
+}
+
+impl<T: 'static + Send + Sync, D: 'static + Send + Sync> AsyncHook
+    for PreviewHighlightHandler<T, D>
+{
+    type Event = Arc<Path>;
+
+    fn handle_event(
+        &mut self,
+        path: Self::Event,
+        timeout: Option<tokio::time::Instant>,
+    ) -> Option<tokio::time::Instant> {
+        if self
+            .trigger
+            .as_ref()
+            .is_some_and(|trigger| trigger == &path)
+        {
+            // If the path hasn't changed, don't reset the debounce
+            timeout
+        } else {
+            self.trigger = Some(path);
+            Some(Instant::now() + Duration::from_millis(250))
```

Comment:
I am surprised you put the timeout so high. @archseer actually said that he would like to lower the idle timeout so that the picker highlights faster and just doens't because of auto completions. Since they are decoupled with this pr we could set this lower. The current value is 200ms I woul at least not set it any higher. I think 150 would be ok too. This is mostly meant to prevent highlighting files while scrolling